packages/button/src/index.tsx

```diff
@@ -38,49 +45,76 @@ const StyledText = styled(Text)<StyledTextProps>`
           display: 'inline-flex'
         }}
   ${_ => getButtonStyle(_.appearance, _.intent)};
+  ${_ => getIconAttachmentStyle(_.height, _.iconPosition)};
 `
 const StyledSpinner = styled(Spinner)<StyledSpinnerProps>`
   opacity: 0.6;
   color: ${_ => _.theme.scales.neutral.N7A};
-  margin-right: ${_ => _.marginRight}px;
-  margin-left: ${_ => _.marginLeft}px;
+  margin-right: ${_ =>
+    _.iconPosition === 'center'
+      ? 0
+      : _.iconPosition === 'left'
+      ? Math.round(_.height / 4)
+      : -Math.round(_.height / 8)}px;
+  margin-left: ${_ =>
+    _.iconPosition === 'center'
+      ? 0
+      : _.iconPosition === 'left'
+      ? -Math.round(_.height / 8)
+      : Math.round(_.height / 4)}px;
   vertical-align: middle;
 `
 
 const ButtonFCFactory: <AdditionalProps extends {}>(
   as: StyledComponentElement
 ) => React.FC<ButtonProps> = <AdditionalProps extends {}>(
   as: StyledComponentElement = 'button'
-) => ({children, innerRef, ...props}) => {
-  // eslint-disable-next-line react-hooks/rules-of-hooks
-  const defaults = useDefaults('button', props, {
-    height: 32,
-    appearance: 'default' as ButtonAppearanceType,
-    intent: 'none' as ButtonIntentType,
-    isLoading: false,
-    full: false
-  })
+) => {
+  const ButtonFC = ({children, innerRef, ...props}) => {
+    const defaults = useDefaults('button', props, {
+      height: 32,
+      appearance: 'default' as ButtonAppearanceType,
+      intent: 'none' as ButtonIntentType,
+      isLoading: false,
+      full: false,
+      iconPosition: 'right' as ButtonIconPosition
+    })
+
+    const theme = useTheme()
+
+    const IconComponent = props.icon
+
+    return (
+      <StyledText
+        as={as}
+        borderRadius={getBorderRadiusForControlHeight(defaults.height)}
+        variant={getTextSizeForControlHeight(defaults.height)}
+        ref={innerRef}
+        {...defaults}
+        {...props}
+        disabled={props.disabled || props.isLoading}
+      >
+        {children}
+        {defaults.isLoading && (
+          <StyledSpinner
+            iconPosition={props.iconPosition}
+            height={defaults.height}
+            size={Math.round(defaults.height / 2)}
+          />
+        )}
+        {IconComponent && !defaults.isLoading && (
+          <div className="svg-wrapper">
```

Comment:
We should remain consistent and use styled components everywhere, no bare classes.